src/main/java/com/linecorp/armeria/server/tracing/HttpTracingServiceInvocationHandler.java

```diff
@@ -40,11 +41,11 @@
     @Override
     protected TraceData getTraceData(ServiceInvocationContext ctx) {
         final Object request = ctx.originalRequest();
-        if (request == null || !(request instanceof HttpRequest)) {
+        if (!(request instanceof HttpRequest)) {
             return TraceData.builder().build();
         }
 
-        final HttpHeaders headers = ((HttpRequest) request).headers();
+        final HttpHeaders headers = ((HttpMessage) request).headers();
```

Comment:
Yes and no. Casting to `HttpRequest` is overly narrow downcast. I'm fine with either but I just wanted to make the IDEA inspector happy.